connect-inject/handler.go

```diff
@@ -399,85 +315,53 @@ func (h *Handler) Mutate(req *v1beta1.AdmissionRequest) *v1beta1.AdmissionRespon
 	//	}
 	//}
 
-	patches = append(patches, addContainer(
-		pod.Spec.Containers,
-		[]corev1.Container{esContainer},
-		"/spec/containers")...)
-
-	// Add annotations so that we know we're injected
-	patches = append(patches, updateAnnotation(
-		pod.Annotations,
-		map[string]string{
-			annotationStatus: injected,
-		})...)
+	pod.Annotations[annotationStatus] = injected
```

Comment:
Do we need a nil check here for `pod.Annotations`?

Reply:
Added a comment here!!